diff --git a/js/private/coverage/BUILD.bazel b/js/private/coverage/BUILD.bazel
index 419a14b4..d0f49396 100644
--- a/js/private/coverage/BUILD.bazel
+++ b/js/private/coverage/BUILD.bazel
@@ -15,5 +15,6 @@ write_source_files(
 
 coverage_merger(
     name = "merger",
+    entry_point = "@bazel_coverage_not_working_repro//src/coverage_lib:coverage.js",
     visibility = ["//visibility:public"],
 )
diff --git a/js/private/js_binary.bzl b/js/private/js_binary.bzl
index b683ecad..320da2b1 100644
--- a/js/private/js_binary.bzl
+++ b/js/private/js_binary.bzl
@@ -571,6 +571,11 @@ def _js_binary_impl(ctx):
         # We have to propagate _lcov_merger runfiles since bazel does not treat _lcov_merger as a proper tool.
         # See: https://github.com/bazelbuild/bazel/issues/4033
         runfiles = runfiles.merge(ctx.attr._lcov_merger[DefaultInfo].default_runfiles)
+        for data in ctx.attr.data:
+            if InstrumentedFilesInfo in data:
+                runfiles = runfiles.merge(ctx.runfiles(
+                    files = data[InstrumentedFilesInfo].instrumented_files.to_list(),
+                ))
         providers = [
             coverage_common.instrumented_files_info(
                 ctx,
